### ISSUE TITLE
Add finer control of terms used for caculating the alias matrix

### ISSUE THREAD

**Issue by StevenCHowell** [tylermorganwall/skpr#48]
I am interested in designing an alias optimal experiment that considers the aliasing with only some of the 2-factor interactions.  

Currently, the `aliaspower` parameter of `gen_design` provides a coarse control of how many powers to include in the alias optimal calculation.  Ideally, one would be able to select specific interactions to include.

Perhaps it makes sense to use a formula, similar to the model parameter.  Something like the following

```r
quad_candidates = expand.grid(a=c(1,0,-1), b=c(-1,0,1), c=c("A","B","C"))
gen_design(
    candidateset = quad_candidates, 
    model = ~a + b + I(a^2) + I(b^2) + c, ),
    optimality = 'Alias',
    alias_model = ~a:b + b:c  # do not worry about aliasing with a:c
)
```

For this example, the alias matrix for the generated design should only evaluate aliasing with the `a:b` and `b:c` interactions, disregarding the `a:c` interactions.

**Comment by tylermorganwall** [comment]
I can look into implementing such an option into the `advancedoptions` argument in `gen_design`. If you're truly concerned about the `a:b` and `b:c` interactions, however, I'd just include them in your model. By definition, the Alias-optimality criterion is specifically meant to minimize the potential bias associated with all terms not in the model. Unless you are absolutely 100% sure that the `a:c` interaction isn't active, it doesn't hurt that much to keep it in the calculation.

**Comment by StevenCHowell** [comment]
With only three examples, adding one more higher order interaction to the optimization certainly would not make much difference.  The real problem I want to apply this to has around 10 factors, and I only expect about 10 of the higher order interactions could possibly be significant.  I would expect including all 100 in the alias optimization would be detrimental toward the 10 I actually care about.

**Comment by tylermorganwall** [comment]
Makes sense, I merged the branch in 03539e1010aa39df03b794a9b838da9bdea435d4 with some minor name changes.
